### DISCUSSION TITLE
How to build docs in GitHub without GitHub Actions or other CI/CD tools

### DISCUSSION THREAD

**Discussion by syeda-git** [squidfunk/mkdocs-material#3088] [Q&A]
Hello,

I am evaluating tools for my company. MkDocs has many advantages. One of the use cases is to edit the source files in GitHub itself  and when you commit the changes, the GitHub Pages site is refreshed. So, GitHub automatically builds the changes and deploys it to GitHub pages in the background. My company's GitHub does not yet support GitHub Actions and we don't want invest time and effort now at setting up a CI/CD tool.

I heard that with Jekyll the above use case is possible. Maybe because GitHub Pages is also based on Jekyll. Is there a way to achieve this capability with MkDocs or MkDocs with Material?

I read these docs:
https://www.mkdocs.org/user-guide/deploying-your-docs/
https://squidfunk.github.io/mkdocs-material/publishing-your-site/

Let me know if you have questions.

Thanks...
Syed

**Comment by squidfunk** [comment]
If you don't want to use GitHub Actions, and still want to build on every commit, you need some other CI/CD tool (Jenkins, Circle CI, etc.) which you said you don't want to invest the time in managing. Jekyll seems to be supported by GitHub Pages, but MkDocs isn't, so there's no way around a CI/CD tool. The only other option I can think of would be to [deploy from the command line](https://squidfunk.github.io/mkdocs-material/publishing-your-site/#with-mkdocs) via a git hook every time you push to a specific branch. Note that this will not catch edits on GitHub. IMHO there's no way around a CI/CD tool.

**Reply by squidfunk** [reply]
Also, please don't double post in the future: https://github.com/mkdocs/mkdocs/discussions/2599

**Comment by NicolasMassart** [comment]
If you want to build mkdocs + material site without CI you can still use ReadTheDocs.org (or .com if you are not an open source project) and it will build and publish automatically without Actions or other CI. It's a bit limited IMO, but if that's your need...\
